From ebffbce8003c7d6a177a2412014e7090476e33a7 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <jouni@qca.qualcomm.com>
Date: Mon, 26 Mar 2012 22:06:48 +0300
Subject: [PATCH 59/73] P2P: Validate p2p_oper_channel in p2p_group_add

If the p2p_group_add command does not specify the operating channel,
make sure the operating channel set in the configuration file meets
the P2P requirements in the same way as is done with the frequency
specified as the command parameter.

Signed-hostap: Jouni Malinen <jouni@qca.qualcomm.com>
(cherry picked from commit f4329aa2d08192640532b712936f5221580e9f8c)
---
 wpa_supplicant/p2p_supplicant.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -3239,6 +3239,13 @@ int wpas_p2p_group_add(struct wpa_suppli
 
 	if (wpas_p2p_init_go_params(wpa_s, &params, freq))
 		return -1;
+	if (params.freq &&
+	    !p2p_supported_freq(wpa_s->global->p2p, params.freq)) {
+		wpa_printf(MSG_DEBUG, "P2P: The selected channel for GO "
+			   "(%u MHz) is not supported for P2P uses",
+			   params.freq);
+		return -1;
+	}
 	p2p_go_params(wpa_s->global->p2p, &params);
 	params.persistent_group = persistent_group;
 
